### PR TITLE
availability and latency slos

### PR DESCRIPTION
**Pull request by R-Lawton**
Creating SLO for both availability and latency using sloth. 

##Verification 
#### Unit tests
Run promtool tests:
`make alerts-tests`
You should see success for both alerts
#### Sloth generation
Install Sloth SLI https://sloth.dev/introduction/install/
Run `generate -i examples/alerts/sloth/latency.yaml --default-slo-period=28d`
The generated yaml should match the current alert for latency and when you change the file to availability you should see the same.

### CODE REVIEW

File: examples/alerts/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+### Sloth
+Sloth is a tool to aid in the creation of multi burn rate and multi window SLO alerts and was used to create both the availability and latency alerts. It follows the common standard set out by [Google's SRE book](https://sre.google/workbook/implementing-slos/). Sloth generates alerts based on specific specs given. The specs for our example alerts can be found in the example/sloth folder.
+
+#### Metrics used for the alerts
+ 
+#### Availability
+For the availability SLO alerts the Istio metric `istio_requests_total` was used as its a counter type metric meaning the values can only increase as well as it gives information on all requests handled by the Istio proxy.
+
+#### Latency
+For the availability SLO alerts the Istio metric 'istio_requests_total' was used as its a Distribution type metric meaning values are mapped to different ranges of frequency's as well as it gives information duration of requests.
```
Comment (david-martin) [inline examples/alerts/README.md:21]:
```suggestion
For the availability SLO alerts the Istio metric `istio_request_duration_milliseconds` was used as its a [Histogram](https://prometheus.io/docs/concepts/metric_types/#histogram) type metric meaning values are mapped to different ranges of frequency's as well as it gives information duration of requests.
```
- metric name fix
- would histogram be a better term here?

File: examples/alerts/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+## SLO Multi burn rate multi window alerts
+Kuadrant have created two example SLO alerts to help give ideas on the types of SLO alerts that could be used with the operator. We have created one alert for latency and one for availability, both are Multiwindow, Multi-Burn-Rate Alerts. The alerts show a scenario where a 28d rolling window is used and a uptime of 99.95% i.e only .1% error margin is desired. This in real world time would be downtime of around. These values can be changed to suit different scenarios
```
Comment (david-martin) [inline examples/alerts/README.md:2]:
```suggestion
Kuadrant have created two example SLO alerts to help give ideas on the types of SLO alerts that could be used with the operator. We have created one alert for latency and one for availability, both are Multiwindow, Multi-Burn-Rate Alerts. The alerts show a scenario where a 28d rolling window is used and a uptime of 99.95% i.e only 0.05% error budget. This in real world time would be downtime of around 21m per month. These values can be changed to suit different scenarios
```
- 0.05% of budget 
- using term 'error budget' instead of margin
- missing time

File: examples/alerts/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+For the availability SLO alerts the Istio metric `istio_requests_total` was used as its a counter type metric meaning the values can only increase as well as it gives information on all requests handled by the Istio proxy.
+
+#### Latency
+For the availability SLO alerts the Istio metric 'istio_requests_total' was used as its a Distribution type metric meaning values are mapped to different ranges of frequency's as well as it gives information duration of requests.
+
+### Sloth generation
+You can modify the examples Sloth specs we have and regenerate the prometheus rules using the Sloth CLI and the generate command. For more information please the [Sloth website](https://sloth.dev/usage/cli/)
+
+```
+generate -i examples/alerts/sloth/latency.yaml --default-slo-period=28d
```
Comment (david-martin) [inline examples/alerts/README.md:27]:
```suggestion
sloth generate -i examples/alerts/sloth/latency.yaml --default-slo-period=28d
```

File: make/alerts.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                                            echo "docker"; \
+                                        elif command -v podman &>/dev/null; then \
+                                            echo "podman"; \
+                                        else \
+                                            echo "Neither Docker nor Podman is installed. Exiting..."; \
+                                            exit 1; \
+                                        fi))
+
+
+alerts-tests: container-runtime-tool
```
Comment (david-martin) [inline make/alerts.mk:20]: would be really nice to have a target that regens the rules.
i.e runs `sloth generate ...`

File: examples/alerts/tests/availability_test.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+rule_files:
+  - /prometheus/availability-rules.yaml
+
+evaluation_interval: 1m
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: istio_requests_total{job="ingress-metrics-proxy",response_code="500"}
+        values: "0+0x30 0+10x30"
```
Comment (david-martin) [inline examples/alerts/tests/availability_test.yaml:10]: i've always found this syntax to be complex to decode.
I wonder if we can add comments to explain in simpler terms what's going on.
Something like:

```yaml
# 0+0x30 => 0, 0, 0, ...0 (total 31 entries)
# then,
# 0+10x30 => 0, 10, 20, ...300 (total 31 entries) 
```

for below, might look like:

```yaml
# 0+1x30 => 0, 1, 2, ...30 (total 31 entries)
# then
# 31+100x30 => 31, 131, 231,...3031 (total 31 entries)
```

**Review by david-martin — CHANGES_REQUESTED** [review]
Really great to have these examples.
Some suggestions inline.

File: examples/alerts/README.md (hunk 1 above)
Comment (R-Lawton) [reply examples/alerts/README.md:21]: Opps copy and paste error. Yeah histogram could work to

File: make/alerts.mk (hunk 4 above)
Comment (R-Lawton) [reply make/alerts.mk:20]: Ooo yeah good shout ill create that

**Review by david-martin — APPROVED** [review]
/lgtm
/approve
